### ISSUE TITLE
[css-will-change] ambiguous phrasing

### ISSUE THREAD

**Issue by frivoal** [w3c/csswg-drafts#4612]
https://drafts.csswg.org/css-will-change/#valdef-will-change-custom-ident has phrasing like (the parts between brackets change in the various related clauses):

> If any non-initial value of a property would [create a stacking context on the element], specifying that property in will-change must [create a stacking context on the element].

I think this use of *any* is ambiguous: if some but not all of the non-initial values of that property would do the thing mentioned, are we in the must case or not?

I suggest rephrasing to either:
- if all values other than the initial value of a property would…
- if at least one of the values other than the initial value of a property would…

**Comment by Loirooriol** [comment]
I always assumed it was the 2nd, e.g. `will-change: contain` would create a stacking context because `contain: layout` does that, even if `contain: size` does not.

That's what Firefox does, but not Chromium. [Testcase](https://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0Adiv%3A%3Abefore%2C%20div%3A%3Aafter%20%7B%0A%20%20display%3A%20inline-block%3B%0A%20%20width%3A%20100%25%3B%0A%20%20vertical-align%3A%20top%3B%0A%20%20height%3A%2025px%3B%0A%7D%0Adiv%3A%3Aafter%20%7B%0A%20%20margin-top%3A%20-25px%3B%0A%7D%0A%0A%23contain-layout%3A%3Abefore%20%7B%0A%20%20content%3A%20%22PASS%20-%20%27contain%3A%20layout%27%20created%20a%20stacking%20context%22%3B%0A%20%20contain%3A%20layout%3B%0A%20%20background%3A%20lime%3B%0A%7D%0A%23contain-layout%3A%3Aafter%20%7B%0A%20%20content%3A%20%22FAIL%20-%20%27contain%3A%20layout%27%20should%20have%20created%20a%20stacking%20context%22%3B%0A%20%20background%3A%20red%3B%0A%7D%0A%0A%23contain-size%3A%3Abefore%20%7B%0A%20%20content%3A%20%22FAIL%20-%20%27contain%3A%20size%27%20shouldn%27t%20have%20created%20a%20stacking%20context%22%3B%0A%20%20contain%3A%20size%3B%0A%20%20background%3A%20red%3B%0A%7D%0A%23contain-size%3A%3Aafter%20%7B%0A%20%20content%3A%20%22PASS%20-%20%27contain%3A%20size%27%20didn%27t%20create%20a%20stacking%20context%22%3B%0A%20%20background%3A%20lime%3B%0A%7D%0A%0A%23will-change-contain%3A%3Abefore%20%7B%0A%20%20content%3A%20%22%3F%3F%20-%20%27will-change%3A%20contain%27%20DID%20create%20a%20stacking%20context%22%3B%0A%20%20will-change%3A%20contain%3B%0A%20%20background%3A%20orange%3B%0A%7D%0A%23will-change-contain%3A%3Aafter%20%7B%0A%20%20content%3A%20%22%3F%3F%20-%20%27will-change%3A%20contain%27%20did%20NOT%20create%20a%20stacking%20context%22%3B%0A%20%20background%3A%20orange%3B%0A%7D%0A%0A%23will-change-transform%3A%3Abefore%20%7B%0A%20%20content%3A%20%22PASS%20-%20%27will-change%3A%20transform%27%20created%20a%20stacking%20context%22%3B%0A%20%20will-change%3A%20transform%3B%0A%20%20background%3A%20lime%3B%0A%7D%0A%23will-change-transform%3A%3Aafter%20%7B%0A%20%20content%3A%20%22FAIL%20-%20%27will-change%3A%20transform%27%20should%20have%20created%20a%20stacking%20context%22%3B%0A%20%20background%3A%20red%3B%0A%7D%0A%3C%2Fstyle%3E%0A%3Cdiv%20id%3D%22contain-layout%22%3E%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22contain-size%22%3E%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22will-change-contain%22%3E%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22will-change-transform%22%3E%3C%2Fdiv%3E).

Not sure if there are other properties like this.

**Comment by mrego** [comment]
@Loirooriol yeah that's a "known issue", the following test fails in Chromium http://w3c-test.org/css/vendor-imports/mozilla/mozilla-central-reftests/contain/contain-paint-stacking-context-001b.html

This issue comes from a question to @frivoal. If this gets clarified, I'll try to fix the behavior in Chromium.

**Comment by ewilligers** [comment]
> I always assumed it was the 2nd, e.g. will-change: contain would create a
> stacking context because contain: layout does that, even if contain: size
> does not.
>
>
I similarly read the sentence that way.

**Comment by tabatkins** [comment]
Yes, I'm not sure why this is ambiguous. "Any" means "at least one of the set". I'm not sure how it could be read as meaning "all". That's the exact opposite meaning of the word; "any" vs "all" is a common opposing pair used throughout specifications.
